spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe 'GET #index' do
+    it 'assigns all users as @users' do
+      user = User.create!(name: 'John Doe', posts_counter: 0)
+      get :index
+      expect(assigns(:users)).to eq([user])
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when the user exists' do
+      it 'assigns the requested user as @user' do
+        user = User.create!(name: 'John Doe', posts_counter: 0)
+        get :show, params: { id: user.to_param }
+        expect(assigns(:user)).to eq(user)
+      end
+    end
+
+    context 'when the user does not exist' do
+      it 'redirects to the users list' do
+        get :show, params: { id: -1 }
+        expect(response).to redirect_to(users_path)
+      end
+    end
+  end
+end
```

Comment:
| Project Requirements 📝 |
|-|
| ![image](https://github.com/fuadmoin/Blog-app/assets/93039359/601cfa21-c099-44f7-88c1-39b092274cfb) |


- [ ] It would be nice if you follow the project requirements because, in the project requirements, it is clearly mentioned that you have to check the things shown in the image above ⬆️ 👍

Hint: You can use the following matchers to complete the testing according to the project requirements 👍

```ruby

--- These below are just examples so you need to write your tests in your own way

RSpec.describe 'Users', type: :request do
  describe 'GET /index' do
    before(:example) { get users_path }
    it 'renders http status' do
      expect(response).to have_http_status(200)
    end

    it "renders 'index' template" do
      expect(response).to render_template(:index)
    end

    it 'should return correct placeholder' do
      expect(response.body).to include '<h1>Users#index</h1>'
    end
  end

  describe 'GET /show' do
    before(:example) { get '/users/1' }
    it 'renders http status' do
      expect(response).to have_http_status 200
    end

    it "renders 'show' template" do
      expect(response).to render_template :show
    end

    it 'should return correct placeholder' do
      expect(response.body).to include '<h1>Single User</h1>'
    end
  end
end
```